apis/bigqueryconnection/v1beta1/connection_types.go

```diff
@@ -0,0 +1,336 @@
+	SparkSpec *SparkPropertiesSpec `json:"spark,omitempty"`
+}
+
+// BigQueryConnectionConnectionStatus defines the config connector machine state of BigQueryConnectionConnection
+type BigQueryConnectionConnectionStatus struct {
+	/* Conditions represent the latest available observations of the
+	   object's current state. */
+	Conditions []v1alpha1.Condition `json:"conditions,omitempty"`
+
+	// ObservedGeneration is the generation of the resource that was most recently observed by the Config Connector controller. If this is equal to metadata.generation, then that means that the current reported status reflects the most recent desired state of the resource.
+	ObservedGeneration *int64 `json:"observedGeneration,omitempty"`
+
+	// A unique specifier for the BigQueryConnectionConnection resource in GCP.
+	ExternalRef *string `json:"externalRef,omitempty"`
+
+	// ObservedState is the state of the resource as most recently observed in GCP.
+	ObservedState *BigQueryConnectionConnectionObservedState `json:"observedState,omitempty"`
+}
+
+// BigQueryConnectionConnectionSpec defines the desired state of BigQueryConnectionConnection
+// +kcc:proto=google.cloud.bigquery.connection.v1.Connection
+type BigQueryConnectionConnectionObservedState struct {
+	Aws *AwsPropertiesStatus `json:"aws,omitempty"`
+
+	Azure *AzurePropertiesStatus `json:"azure,omitempty"`
+
+	CloudResource *CloudResourcePropertiesStatus `json:"cloudResource,omitempty"`
+
+	CloudSQL *CloudSqlPropertiesStatus `json:"cloudSQL,omitempty"`
+
+	Spark *SparkPropertiesStatus `json:"spark,omitempty"`
+
+	// The display name for the connection.
+	FriendlyName *string `json:"friendlyName,omitempty"`
+
+	// The description for the connection.
+	Description *string `json:"description,omitempty"`
+
+	// Output only. True, if credential is configured for this connection.
+	HasCredential *bool `json:"hasCredential,omitempty"`
+}
+
+type AwsPropertiesSpec struct {
+	// Authentication using Google owned service account to assume into
+	//  customer's AWS IAM Role.
+	// +required
+	AccessRole *AwsAccessRoleSpec `json:"accessRole,omitempty"`
+}
+
+type AwsAccessRoleSpec struct {
+	// The user’s AWS IAM Role that trusts the Google-owned AWS IAM user
+	//  Connection.
+	// +required
+	IamRoleID *string `json:"iamRoleID,omitempty"`
+}
+
+type AzurePropertiesSpec struct {
+	// The id of customer's directory that host the data.
+	// +required
+	CustomerTenantID *string `json:"customerTenantID,omitempty"`
+
+	// The client ID of the user's Azure Active Directory Application used for a
+	//  federated connection.
+	FederatedApplicationClientID *string `json:"federatedApplicationClientID,omitempty"`
+}
+
+type CloudResourcePropertiesSpec struct{}
+
+type CloudSqlPropertiesSpec struct {
+	// Reference to the Cloud SQL instance ID.
+	// +required
+	InstanceRef *refv1beta1.SQLInstanceRef `json:"instanceRef,omitempty"`
+
+	// Database name.
+	// +required
+	Database *string `json:"database,omitempty"`
```

Comment:
We just realized that this field should be a reference field to `SQLDatabase` resource. Could you help fix it?